Fix UNIX Socket tests on OS X

UNIX Socket tests are broken on OS X because the ``tmpdir`` fixture that [pytest](https://pypi.python.org/pypi/pytest) provides creates paths that are apparently too long:
```#!bash
(circuits)
prologic@Jamess-MacBook
Sun Jan 17 20:12:27
~/circuits
 (master) 0
$ py.test -x -s -v tests/net/test_unix.py
===================================================== test session starts ======================================================
platform darwin -- Python 2.7.11, pytest-2.8.5, py-1.4.31, pluggy-0.3.1 -- /Users/prologic/.virtualenvs/circuits/bin/python2.7
cachedir: .cache
rootdir: /Users/prologic/work/circuits, inifile:
plugins: cov-2.2.0
collected 2 items

tests/net/test_unix.py::test_unix[0] FAILED

=========================================================== FAILURES ===========================================================
_________________________________________________________ test_unix[0] _________________________________________________________

tmpdir = local('/private/var/folders/37/y3fcncvx1yj5dy4xsk5qw_n80000gn/T/pytest-of-prologic/pytest-14/test_unix_0_0')
Poller = <class 'circuits.core.pollers.Select'>

    def test_unix(tmpdir, Poller):
        m = Manager() + Poller()

        sockpath = tmpdir.ensure("test.sock")
        filename = str(sockpath)

>       server = Server() + UNIXServer(filename)

tests/net/test_unix.py:40:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
circuits/net/sockets.py:402: in __init__
    self._sock = self._create_socket()
circuits/net/sockets.py:712: in _create_socket
    sock.bind(self._bind)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'bind', self = <socket._socketobject object at 0x10a616d70>
args = ('/private/var/folders/37/y3fcncvx1yj5dy4xsk5qw_n80000gn/T/pytest-of-prologic/pytest-14/test_unix_0_0/test.sock',)

    def meth(name,self,*args):
>       return getattr(self._sock,name)(*args)
E       error: AF_UNIX path too long

/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py:228: error
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=================================================== 1 failed in 0.22 seconds ===================================================
```

--- a/tests/net/test_unix.py
+++ b/tests/net/test_unix.py
@@ -1,21 +1,35 @@
 #!/usr/bin/env python
 
 import pytest
+from pytest import fixture
+
 
 import os
 import sys
+import tempfile
 import select
 
-if sys.platform in ("win32", "cygwin"):
-    pytest.skip("Test Not Applicable on Windows")
 
 from circuits import Manager
 from circuits.net.sockets import close, connect, write
 from circuits.net.sockets import UNIXServer, UNIXClient
 from circuits.core.pollers import Select, Poll, EPoll, KQueue
 
+
 from .client import Client
 from .server import Server
+
+
+if sys.platform in ("win32", "cygwin"):
+    pytest.skip("Test Not Applicable on Windows")
+
+
+@fixture
+def tmpfile(request):
+    tmpdir = tempfile.mkdtemp()
+    filename = os.path.join(tmpdir, "test.sock")
+
+    return filename
 
 
 def pytest_generate_tests(metafunc):
@@ -31,13 +45,10 @@ def pytest_generate_tests(metafunc):
         metafunc.addcall(funcargs={"Poller": KQueue})
 
 
-def test_unix(tmpdir, Poller):
+def test_unix(tmpfile, Poller):
     m = Manager() + Poller()
 
-    sockpath = tmpdir.ensure("test.sock")
-    filename = str(sockpath)
-
-    server = Server() + UNIXServer(filename)
+    server = Server() + UNIXServer(tmpfile)
     client = Client() + UNIXClient()
 
     server.register(m)
@@ -49,7 +60,7 @@ def test_unix(tmpdir, Poller):
         assert pytest.wait_for(server, "ready")
         assert pytest.wait_for(client, "ready")
 
-        client.fire(connect(filename))
+        client.fire(connect(tmpfile))
         assert pytest.wait_for(client, "connected")
         assert pytest.wait_for(server, "connected")
         assert pytest.wait_for(client, "data", b"Ready")
@@ -65,4 +76,3 @@ def test_unix(tmpdir, Poller):
         assert pytest.wait_for(server, "closed")
     finally:
         m.stop()
-        os.remove(filename)

